decidim-core/app/controllers/decidim/user_activities_controller.rb

```diff
@@ -23,29 +23,48 @@ def user
     end
 
     def activities
-      @activities ||= paginate(
-        PublicActivities.new(
-          current_organization,
-          user: user,
-          current_user: current_user,
-          resource_name: filter.resource_type
-        ).query.with_resource_type("all")
-      )
+      @activities ||= begin
```

Comment:
Maybe a bit more clearer if we swap the `begin..end` block with the `paginate` function?

```ruby
@activities ||= paginate(
  if own_activities?
    OwnActivities.new(current_organization, **activity_options).query.with_private_resources
  else
    PublicActivities.new(current_organization, **activity_options).query.with_all_resources
  end
)
```